chain/x/bridge/types/msgs.go

```diff
@@ -0,0 +1,183 @@
+func (msg MsgUpdateChainID) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// GetSignBytes implements the sdk.Msg interface for MsgUpdateChainID.
+func (msg MsgUpdateChainID) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// MsgUpdateValidators is a message for update chain ID of BandChain.
+type MsgUpdateValidators struct {
+	Validators []tmtypes.Validator
+	Sender     sdk.AccAddress
+}
+
+// NewMsgUpdateValidators creates a new MsgUpdateValidators instance.
+func NewMsgUpdateValidators(
+	validators []tmtypes.Validator,
+	sender sdk.AccAddress,
+) MsgUpdateValidators {
+	return MsgUpdateValidators{
+		Validators: validators,
+		Sender:     sender,
+	}
+}
+
+// Route implements the sdk.Msg interface for MsgUpdateValidators.
+func (msg MsgUpdateValidators) Route() string { return RouterKey }
+
+// Type implements the sdk.Msg interface for MsgUpdateValidators.
+func (msg MsgUpdateValidators) Type() string { return "update_validators" }
+
+// ValidateBasic implements the sdk.Msg interface for MsgUpdateValidators.
+func (msg MsgUpdateValidators) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(msg.Sender); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "sender: %s", msg.Sender)
+	}
+	return nil
+}
+
+// GetSigners implements the sdk.Msg interface for MsgUpdateValidators.
+func (msg MsgUpdateValidators) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// GetSignBytes implements the sdk.Msg interface for MsgUpdateValidators.
+func (msg MsgUpdateValidators) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// MsgRelay is a message for relay block on BandChain.
+type MsgRelay struct {
+	SignedHeader tmtypes.SignedHeader
+	Sender       sdk.AccAddress
+}
+
+// NewMsgRelay creates a new MsgRelay instance.
+func NewMsgRelay(
+	signedHeader tmtypes.SignedHeader,
+	sender sdk.AccAddress,
+) MsgRelay {
+	return MsgRelay{
+		SignedHeader: signedHeader,
+		Sender:       sender,
+	}
+}
+
+// Route implements the sdk.Msg interface for MsgRelay.
+func (msg MsgRelay) Route() string { return RouterKey }
+
+// Type implements the sdk.Msg interface for MsgRelay.
+func (msg MsgRelay) Type() string { return "relay" }
+
+// ValidateBasic implements the sdk.Msg interface for MsgRelay.
+func (msg MsgRelay) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(msg.Sender); err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "sender: %s", msg.Sender)
+	}
+	return nil
+}
+
+// GetSigners implements the sdk.Msg interface for MsgRelay.
+func (msg MsgRelay) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// GetSignBytes implements the sdk.Msg interface for MsgRelay.
+func (msg MsgRelay) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))
+}
+
+// MsgVerifyProof is a message for verify proof
+type MsgVerifyProof struct {
```

Comment:
I think we need to receive block height of this proof for get app hash from store